components/raftstore/src/store/peer_storage.rs

```diff
@@ -201,10 +207,36 @@ impl EntryCache {
             .drain(..(cmp::min(cache_last_idx + 1, idx) - cache_first_idx) as usize);
         if self.cache.len() < SHRINK_CACHE_CAPACITY && self.cache.capacity() > SHRINK_CACHE_CAPACITY
         {
+            let current_mem_size = self.get_mem_size();
             // So the peer storage doesn't have much writes since the proposal of compaction,
             // we can consider this peer is going to be inactive.
             self.cache.shrink_to_fit();
+            self.update_mem_size_change(current_mem_size, self.get_mem_size());
+        }
+    }
+
+    fn get_mem_size(&self) -> i64 {
+        let data_size: usize = self
```

Comment:
The purpose of storing memory size delta is to avoid scanning the cache, which is `O(n)` complexity.

Reply:
I'm so stupid, sorry for wasting your time. How about now?